src/asm/opt.c

```diff
@@ -71,16 +78,31 @@ void opt_Parse(char *s)
 			error("Must specify exactly 4 characters for option 'g'\n");
 		break;
 
+	case 'f':
+		if (strlen(&s[1]) <= 2) {
+			int result;
+			unsigned int precision;
+
+			result = sscanf(&s[1], "%u", &precision);
+			if ((result != EOF && result != 1) || precision < 1 || precision > 31)
```

Comment:
> The value **EOF** is returned if the end of input is reached before either the first successful conversion or a matching failure occurs.  **EOF** is also returned if a read error occurs, in which case the error indicator for the stream (see **ferror(3)**) is set, and errno is set to indicate the error.

So getting `EOF` back should be an error, if only since failure to read `precision` means its value is uninit'd.

Reply:
Again, this is based on the condition for `fillByte`; should they both use `||` instead of `&&`?

Reply:
No, just `result != 1`